### ISSUE TITLE
Client 1 quit with exit code 1 and signal 0

### ISSUE THREAD

**Issue by yashguptaz** [neovim/nvim-lspconfig#891]
- `nvim --version`:
NVIM v0.5.0-dev+1302-g8a93d1028
 
- nvim-lsp version(commit hash):
0a921bf2be74d293c8a8d6dd70122c1d01151a30

- What language server (If the problem is related to a specific language server):
TSServer

- Can you reproduce this behavior on other language server clients (vscode, languageclient-neovim, coc.nvim, etc.):
I don't know how to.

- Can you reproduce this behavior on other language servers offered in the nvim-lspconfig repo? (pyls -> pyright):
Yes, the problem is reproducible with pyright.

- Is the problem isolated to a particular language server: No
- Operating system/version: Windows 10

### How to reproduce the problem from neovim startup
As soon as I open a TypeScript or a Python file I get this issue.

### Actual behaviour
I get `Client 1 quit with exit code 1 and signal 0` in `:messages`

### Expected behaviour
Language Server should work

### Minimal init.vim or init.lua and code sample

The only changes are on line 50 & 51
```lua
  local name = "tsserver"
  local cmd = { "typescript-language-server.cmd", "--stdio"} -- needed for elixirls, omnisharp, sumneko_lua
```

### Health check
<details>
<summary>Checkhealth result</summary>

health#lspconfig#check
========================================================================
## Checking language server protocol configuration
  - INFO: pyright: configuration checked.
  - INFO: tsserver: configuration checked.
</details>

### LSP log

<details>
<summary>Log file</summary>

</details>

**Comment by mjlbach** [comment]
That's a very generic error message, it just means the the language server quit unexpectedly. You didn't post your log file, make sure you can launch tsserver (which also requires typescript to be installed as per CONFIG.md)/your python language server from the command line

**Comment by yashguptaz** [comment]
<details>
<summary>Log file</summary>
[ START ] 2021-05-09T01:11:27+0530 ] LSP logging initiated
[ START ] 2021-05-09T01:11:40+0530 ] LSP logging initiated
[ INFO ] 2021-05-09T01:11:45+0530 ] ...m-nightly\current\share\nvim\runtime\lua\vim\lsp\rpc.lua:311 ]	"Starting RPC client"	{  args = { "--stdio" },  cmd = "typescript-language-server.cmd",  extra = {}}
[ DEBUG ] 2021-05-09T01:11:45+0530 ] ...eovim-nightly\current\share\nvim\runtime\lua\vim\lsp.lua:826 ]	"LSP[tsserver]"	"initialize_params"	{  capabilities = {    callHierarchy = {      dynamicRegistration = false,      <metatable> = <1>{        __tostring = <function 1>      }    },    textDocument = {      codeAction = {        codeActionLiteralSupport = {          codeActionKind = {            valueSet = { "", "Empty", "QuickFix", "Refactor", "RefactorExtract", "RefactorInline", "RefactorRewrite", "Source", "SourceOrganizeImports", "quickfix", "refactor", "refactor.extract", "refactor.inline", "refactor.rewrite", "source", "source.organizeImports" },            <metatable> = <table 1>          },          <metatable> = <table 1>        },        dynamicRegistration = false,        <metatable> = <table 1>      },      completion = {        completionItem = {          commitCharactersSupport = false,          deprecatedSupport = false,          documentationFormat = { "markdown", "plaintext" },          preselectSupport = false,          snippetSupport = false,          <metatable> = <table 1>        },        completionItemKind = {          valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 },          <metatable> = <table 1>        },        contextSupport = false,        dynamicRegistration = false,        <metatable> = <table 1>      },      declaration = {        linkSupport = true,        <metatable> = <table 1>      },      definition = {        linkSupport = true,        <metatable> = <table 1>      },      documentHighlight = {        dynamicRegistration = false,        <metatable> = <table 1>      },      documentSymbol = {        dynamicRegistration = false,        hierarchicalDocumentSymbolSupport = true,        symbolKind = {          valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },          <metatable> = <table 1>        },        <metatable> = <table 1>      },      hover = {        contentFormat = { "markdown", "plaintext" },        dynamicRegistration = false,        <metatable> = <table 1>      },      implementation = {        linkSupport = true,        <metatable> = <table 1>      },      publishDiagnostics = {        relatedInformation = true,        tagSupport = {          valueSet = { 1, 2 },          <metatable> = <table 1>        },        <metatable> = <table 1>      },      references = {        dynamicRegistration = false,        <metatable> = <table 1>      },      rename = {        dynamicRegistration = false,        prepareSupport = true,        <metatable> = <table 1>      },      signatureHelp = {        dynamicRegistration = false,        signatureInformation = {          documentationFormat = { "markdown", "plaintext" },          <metatable> = <table 1>        },        <metatable> = <table 1>      },      synchronization = {        didSave = true,        dynamicRegistration = false,        willSave = false,        willSaveWaitUntil = false,        <metatable> = <table 1>      },      typeDefinition = {        linkSupport = true,        <metatable> = <table 1>      },      <metatable> = <table 1>    },    window = {      showDocument = {        support = false,        <metatable> = <table 1>      },      showMessage = {        messageActionItem = {          additionalPropertiesSupport = false,          <metatable> = <table 1>        },        <metatable> = <table 1>      },      workDoneProgress = true,      <metatable> = <table 1>    },    workspace = {      applyEdit = true,      configuration = true,      symbol = {        dynamicRegistration = false,        hierarchicalWorkspaceSymbolSupport = true,        symbolKind = {          valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },          <metatable> = <table 1>        },        <metatable> = <table 1>      },      workspaceEdit = {        resourceOperations = { "rename", "create", "delete" },        <metatable> = <table 1>      },      workspaceFolders = true,      <metatable> = <table 1>    }  },  clientInfo = {    name = "Neovim",    version = "0.5.0"  },  initializationOptions = vim.empty_dict(),  processId = 7568,  rootPath = "C:\\Users\\yashg\\AppData\\Local\\nvim\\test",  rootUri = "file:///C:/Users/yashg/AppData/Local/nvim/test",  trace = "off",  workspaceFolders = { {      name = "C:\\Users\\yashg\\AppData\\Local\\nvim\\test",      uri = "file:///C:/Users/yashg/AppData/Local/nvim/test"    } }}
[ DEBUG ] 2021-05-09T01:11:45+0530 ] ...m-nightly\current\share\nvim\runtime\lua\vim\lsp\rpc.lua:390 ]	"rpc.send.payload"	{  id = 1,  jsonrpc = "2.0",  method = "initialize",  params = {    capabilities = {      callHierarchy = {        dynamicRegistration = false,        <metatable> = <1>{          __tostring = <function 1>        }      },      textDocument = {        codeAction = {          codeActionLiteralSupport = {            codeActionKind = {              valueSet = { "", "Empty", "QuickFix", "Refactor", "RefactorExtract", "RefactorInline", "RefactorRewrite", "Source", "SourceOrganizeImports", "quickfix", "refactor", "refactor.extract", "refactor.inline", "refactor.rewrite", "source", "source.organizeImports" },              <metatable> = <table 1>            },            <metatable> = <table 1>          },          dynamicRegistration = false,          <metatable> = <table 1>        },        completion = {          completionItem = {            commitCharactersSupport = false,            deprecatedSupport = false,            documentationFormat = { "markdown", "plaintext" },            preselectSupport = false,            snippetSupport = false,            <metatable> = <table 1>          },          completionItemKind = {            valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 },            <metatable> = <table 1>          },          contextSupport = false,          dynamicRegistration = false,          <metatable> = <table 1>        },        declaration = {          linkSupport = true,          <metatable> = <table 1>        },        definition = {          linkSupport = true,          <metatable> = <table 1>        },        documentHighlight = {          dynamicRegistration = false,          <metatable> = <table 1>        },        documentSymbol = {          dynamicRegistration = false,          hierarchicalDocumentSymbolSupport = true,          symbolKind = {            valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },            <metatable> = <table 1>          },          <metatable> = <table 1>        },        hover = {          contentFormat = { "markdown", "plaintext" },          dynamicRegistration = false,          <metatable> = <table 1>        },        implementation = {          linkSupport = true,          <metatable> = <table 1>        },        publishDiagnostics = {          relatedInformation = true,          tagSupport = {            valueSet = { 1, 2 },            <metatable> = <table 1>          },          <metatable> = <table 1>        },        references = {          dynamicRegistration = false,          <metatable> = <table 1>        },        rename = {          dynamicRegistration = false,          prepareSupport = true,          <metatable> = <table 1>        },        signatureHelp = {          dynamicRegistration = false,          signatureInformation = {            documentationFormat = { "markdown", "plaintext" },            <metatable> = <table 1>          },          <metatable> = <table 1>        },        synchronization = {          didSave = true,          dynamicRegistration = false,          willSave = false,          willSaveWaitUntil = false,          <metatable> = <table 1>        },        typeDefinition = {          linkSupport = true,          <metatable> = <table 1>        },        <metatable> = <table 1>      },      window = {        showDocument = {          support = false,          <metatable> = <table 1>        },        showMessage = {          messageActionItem = {            additionalPropertiesSupport = false,            <metatable> = <table 1>          },          <metatable> = <table 1>        },        workDoneProgress = true,        <metatable> = <table 1>      },      workspace = {        applyEdit = true,        configuration = true,        symbol = {          dynamicRegistration = false,          hierarchicalWorkspaceSymbolSupport = true,          symbolKind = {            valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },            <metatable> = <table 1>          },          <metatable> = <table 1>        },        workspaceEdit = {          resourceOperations = { "rename", "create", "delete" },          <metatable> = <table 1>        },        workspaceFolders = true,        <metatable> = <table 1>      }    },    clientInfo = {      name = "Neovim",      version = "0.5.0"    },    initializationOptions = vim.empty_dict(),    processId = 7568,    rootPath = "C:\\Users\\yashg\\AppData\\Local\\nvim\\test",    rootUri = "file:///C:/Users/yashg/AppData/Local/nvim/test",    trace = "off",    workspaceFolders = { {        name = "C:\\Users\\yashg\\AppData\\Local\\nvim\\test",        uri = "file:///C:/Users/yashg/AppData/Local/nvim/test"      } }  }}
[ ERROR ] 2021-05-09T01:11:45+0530 ] ...m-nightly\current\share\nvim\runtime\lua\vim\lsp\rpc.lua:457 ]	"rpc"	"typescript-language-server.cmd"	"stderr"	"error: Found argument '--stdio' which wasn't expected, or isn't valid in this context\n\tDid you mean --stdin?\n\nUSAGE:\n    nu.exe --stdin\n\nFor more information try --help\n"
</details>
@mjlbach  Here's my logfile, and I can confirm I have both TypeScript and Python installed along with the corresponding servers.

**Comment by mjlbach** [comment]
```
"file:///C:/Users/yashg/AppData/Local/nvim/test" } } }} [ ERROR ] 2021-05-09T01:11:45+0530 ] ...m-nightly\current\share\nvim\runtime\lua\vim\lsp\rpc.lua:457 ] "rpc" "typescript-language-server.cmd" "stderr" "error: Found argument '--stdio' which wasn't expected, or isn't valid in this context\n\tDid you mean --stdin?\n\nUSAGE:\n nu.exe --stdin\n\nFor more information try --help\n" ```

This seems to be your problem ^

**Comment by mjlbach** [comment]
FWIW, I just installe typescript-language-server with `npm i -g typescript typescript-language-server` and `typescript --stdio` worked fine for me. So I assume it's an error with your installation (possibly a similarly named package or an old version of something is installed)

**Comment by yashguptaz** [comment]
Similar issue for Python
```
[ ERROR ] 2021-05-09T01:19:34+0530 ] ...m-nightly\current\share\nvim\runtime\lua\vim\lsp\rpc.lua:457 ]	"rpc"	"pyright-langserver.cmd"	"stderr"	"error: Found argument '--stdio' which wasn't expected, or isn't valid in this context\n\tDid you mean --stdin?\n\nUSAGE:\n    nu.exe --stdin\n\nFor more information try --help\n"
```

I ran the command `npm i -g typescript typescript-language-server` again :/

I can perfectly execute `typescript-language-server --stdio` on my terminal.
Do you have any instructions on how to try run my typescript-language-server using another client.

Or could this be a possible neovim issue?

**Comment by yashguptaz** [comment]
@mjlbach It's the name of my shell https://www.nushell.sh/ (it is like zsh or fish shell)

**Comment by mjlbach** [comment]
Try with the standard shell, for whatever reason that seems to be the culprit

**Comment by yashguptaz** [comment]
@mjlbach Thank You so much for the help!
It works with the standard cmd shell :D

**Comment by mjlbach** [comment]
No problem, I'm not familiar with nushell, especially on windows. I'm not going to say we don't support it, but I think any investigation into its compatibility with uv.spawn which we use to launch language servers is going to have to be externally motivated :)

**Comment by yashguptaz** [comment]
I found out why I was facing the issue.
Basically, there is a `%COMSPEC%` environment variable in Windows, 
I decided to set it to `nu.exe` (nushell) so that I get it as my default terminal.

If you want to use neovim along with nushell, look for settings in your terminal emulator to change your shell instead of changing the `%COMSPEC%` variable.

I hope this is useful for anyone who faces the same issue as me😅
